### ISSUE TITLE
not base64 encode a secret from keyvault

### ISSUE THREAD

**Issue by chuckthompsonprofisee** [Azure/secrets-store-csi-driver-provider-azure#315]
Is there a way to have the csi driver not base64 encode a value it pulls from the keyvault?  I have a secret in the keyvault that is already base64 encoded, so when the driver pulls the value from the keyvault it base64 encodes the value again.  so when it ends up in the kubernetes secret, itss a base64 encodes string of a base64 encoded string.  is there anyways to have the driver not encode the value it pulls from keyvault?

**Comment by aramase** [comment]
@chuckthompsonprofisee The driver doesn't do the base64 encoding. When data is written in Kubernetes secret it is automatically base64 encoded. The AKV provider for the CSI driver fetches the secret and writes to the mount as is.

You can set `objectEncoding: base64` in the `SecretProviderClass` for the object - https://github.com/Azure/secrets-store-csi-driver-provider-azure#create-your-own-secretproviderclass-object. The AKV provider will base64 decode the string and write to mount. When the same mount content will be mirrored as Kubernetes secret, it'll be base64 encoded and you should have the string.

**Comment by chuckthompsonprofisee** [comment]
Thanks.  I tried using the objectEncoding: "base64" but I got this error.  
MountVolume.SetUp failed for volume "secrets-store-inline" :
  kubernetes.io/csi: mounter.SetupAt failed: rpc error: code = Unknown desc =
  failed to mount secrets store objects for pod
err: rpc error: code = Unknown desc
  = failed to mount objects, error: illegal base64 data at input byte 0

The string in the secret in the vault is base64 though.  any ideas?
